src/garage/torch/policies/tanh_gaussian_mlp_policy.py

```diff
@@ -0,0 +1,196 @@
+
+    """
+
+    def __init__(self,
+                 env_spec,
+                 hidden_sizes=(32, 32),
+                 hidden_nonlinearity=nn.ReLU,
+                 hidden_w_init=nn.init.xavier_uniform_,
+                 hidden_b_init=nn.init.zeros_,
+                 output_nonlinearity=None,
+                 output_w_init=nn.init.xavier_uniform_,
+                 output_b_init=nn.init.zeros_,
+                 init_std=1.0,
+                 min_std=np.exp(-20.),
+                 max_std=np.exp(2.),
+                 std_parameterization='exp',
+                 layer_normalization=False):
+
+        self._obs_dim = env_spec.observation_space.flat_dim
+        self._action_dim = env_spec.action_space.flat_dim
+
+        Policy.__init__(self, env_spec, name='TanhGaussianPolicy')
+        GaussianMLPTwoHeadedModule.__init__(
+            self,
+            input_dim=self._obs_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=hidden_sizes,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            init_std=init_std,
+            min_std=min_std,
+            max_std=max_std,
+            std_parameterization=std_parameterization,
+            layer_normalization=layer_normalization,
+            normal_distribution_cls=TanhNormal)
+
+    def get_action(self, observation):
+        r"""Get a single action given an observation.
+
+        Args:
+            observation (np.ndarray): Observation from the environment.
+                Shape is :math:`env_spec.observation_space`.
+
+        Returns:
+            tuple:
+                * np.ndarray: Predicted action. Shape is
+                    :math:`env_spec.action_space`.
+                * dict:
+                    * np.ndarray[float]: Mean of the distribution.
+                    * np.ndarray[float]: Standard deviation of logarithmic
+                        values of the distribution.
+
+        """
+        with torch.no_grad():
+            if not isinstance(observation, torch.Tensor):
+                observation = torch.from_numpy(observation).to(
+                    tu.global_device())
+            observation = observation.unsqueeze(0)
+            dist = self.forward(observation)
+            ret_mean = dist.mean.squeeze(0).cpu().numpy()
+            ret_log_std = (dist.variance.sqrt()).log().squeeze(0).cpu().numpy()
+            return (dist.rsample().squeeze(0).cpu().numpy(),
+                    dict(mean=ret_mean, log_std=ret_log_std))
+
+    def get_actions(self, observations):
+        r"""Get actions given observations.
+
+        Args:
+            observations (np.ndarray): Observations from the environment.
+                Shape is :math:`batch_dim \bullet env_spec.observation_space`.
```

Comment:
wait -- really? this seems really weird. isn't the shape just `(batch_dim, S^*)`

also, please use the symbols defined in the documentation for _dtypes.py rather than inventing your own or using words like `batch_dim`

Reply:
Wasn't sure how to approach that, thanks for the clarification.

Reply:
Wasn't sure how to approach that, thanks for the clarification.